src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/DeleteToolbarAction.js

```diff
@@ -68,7 +68,7 @@ export default class DeleteToolbarAction extends AbstractFormToolbarAction {
     navigateBack = () => {
         const {backView} = this.router.route.options;
         const {locale} = this.resourceFormStore;
-        this.router.navigate(backView, {locale: locale ? locale.get() : undefined});
+        this.router.restore(backView, {locale: locale ? locale.get() : undefined});
```

Comment:
@danrot if I reload on the detail form of the page. And then remove the page it will error because it can not restore it I think.

Reply:
I see multiple solutions to this problem:

1. We use the [`getDerivedRouteAttributes` function of the `WebspaceTabs`](https://github.com/sulu/sulu/blob/edbf2d5/src/Sulu/Bundle/PageBundle/Resources/js/views/WebspaceTabs/WebspaceTabs.js#L27) to link to a different webspace it is missing. Problem is that we currently have the `WebspaceStore`, which only offers async functions, which is not supported in this function... So we would have to change to a `WebspaceRegistry` without an async workflow, which would be a BC break, and load all webspaces initially (something that we originally wanted to avoid, but I am not sure if that is worth all the hassle...)
2. We blindly add all available options here, but I am afraid of all the possible side effects caused by that.
3. We introduce one of our loved `routerAttributesToBackView` options on this `ToolbarAction` as well.

The first approach would have the nice side effect that other code also gets easier, and the list would always show something, but I think the use case you are describing here could potentially lead to a wrong webspace.

Therefore I would go for the third option. Maybe we should also ask for the opinion of @sulu/core-team, because that's a very generic approach, which might spread further to other `ToolbarAction`s.

Reply:
After a short discussion with @alexander-schranz and @nnatter we've decided to go for option 3, because it is not too much work, and limits the possible side effects.